### DISCUSSION TITLE
Firedrake Matrix Transpose

### DISCUSSION THREAD

**Discussion by karanh97** [firedrakeproject/firedrake#2782] [Firedrake support]
I was hoping to write a function that takes a Firedrake matrix as its argument and returns the transposed matrix as a Firedrake matrix. I have explored the possibility of first converting the input to a PETSc matrix, but I don't think it is possible to convert a PETSc matrix to a Firedrake matrix. Any suggestions are welcome!

**Comment by dham** [comment]
Currently there are only a limited range of things one can actually do with a Firedrake Matrix (this will hopefully change when the last parts of dual spaces land). Can you explain what it is you want to do with the transpose matrix and we might be able to explain how to do that.
